disnake/guild.py

```diff
@@ -560,12 +562,10 @@ def _from_data(self, guild: GuildPayload) -> None:
         self.widget_enabled: Optional[bool] = guild.get("widget_enabled")
         self.widget_channel_id: Optional[int] = utils._get_as_snowflake(guild, "widget_channel_id")
 
-        self._stage_instances: Dict[int, StageInstance] = {}
         for s in guild.get("stage_instances", []):
```

Comment:
I believe that if we have the data, then we should start from scratch, rather than not clearing them at all.

```py
stage_instances = guild.get("stage_instances")
if stage_instances:
    self._stage_instances: Dict[int, StageInstance] = {}
    for s in stage_instances:
        stage_instance = StageInstance(guild=self, data=s, state=state)
        self._stage_instances[stage_instance.id] = stage_instance
```

Reply:
4a5d694e9f8b2144a1aabc0b3168c4d56e29f024